python/cudf/cudf/_libxx/table.pyx

```diff
@@ -227,6 +226,27 @@ cdef class Table:
             self._index._data.columns
         )
 
+    @staticmethod
+    cdef columns_from_ptr(
+        unique_ptr[table] c_tbl,
+    ):
+        """
+        Return a list of table columns from a unique pointer
+
+        Parameters
+        ----------
+        c_tbl : unique_ptr[cudf::table]
+        """
+        num_columns = c_tbl.get().num_columns()
+        cdef vector[unique_ptr[column]] columns
+        columns = c_tbl.get()[0].release()
+        cdef vector[unique_ptr[column]].iterator it = columns.begin()
+
+        result = []
```

Comment:
Since we know the size of result up front we can empty initialize it to avoid having to constantly resize under the hood.
```suggestion
        result = [None] * num_columns
```